src/theory/smt_engine_subsolver.h

```diff
@@ -41,10 +41,13 @@ namespace theory {
  * if the current SMT engine has declared a separation logic heap.
  *
  * @param smte The smt engine pointer to initialize
+ * @param opts The options for the subsolver. If nullptr, then we copy the
+ * options from the current SmtEngine in scope.
  * @param needsTimeout Whether we would like to set a timeout
  * @param timeout The timeout (in milliseconds)
  */
 void initializeSubsolver(std::unique_ptr<SmtEngine>& smte,
+                         Options* opts = nullptr,
```

Comment:
Passing this kinda voids the need for the other two parameters. Maybe we want to simplify the interface? We could have a wrapper around this method which takes just the timeout for the case when one wants to change just the timeout.

Reply:
I'd rather avoid having too many variants of the same method.  Indeed, the interface is slightly redundant but it still allows casual use with timeout (without having to deal with an explicit options object), this comes at the cost of passing an additional `nullptr` but I don't think this is too big of a deal.

Reply:
OK.